data/mods/ssb/scripts.ts

```diff
@@ -100,324 +212,650 @@ export const Scripts: ModdedBattleScriptsData = {
-			} else {
-				if (move.name === item.zMoveFrom) return item.zMove as string;
+		const {targets, pressureTargets} = pokemon.getMoveTargets(move, target);
+		if (targets.length) {
+			target = targets[targets.length - 1]; // in case of redirection
+		}
+
+		if (!sourceEffect || sourceEffect.id === 'pursuit' || sourceEffect.id === 'thehuntison') {
+			let extraPP = 0;
+			for (const source of pressureTargets) {
+				const ppDrop = this.runEvent('DeductPP', source, pokemon, move);
+				if (ppDrop !== true) {
+					extraPP += ppDrop || 0;
+				}
 			}
-		} else if (item.zMove === true) {
-			if (move.type === item.zMoveType) {
-				if (move.category === "Status") {
-					return move.name;
-				} else if (move.zMove?.basePower) {
-					return this.zMoveTable[move.type];
+			if (extraPP > 0) {
+				pokemon.deductPP(move, extraPP);
+			}
+		}
+
+		if (!this.singleEvent('TryMove', move, null, pokemon, target, move) ||
+			!this.runEvent('TryMove', pokemon, target, move)) {
+			move.mindBlownRecoil = false;
+			return false;
+		}
+
+		this.singleEvent('UseMoveMessage', move, null, pokemon, target, move);
+
+		if (move.ignoreImmunity === undefined) {
+			move.ignoreImmunity = (move.category === 'Status');
+		}
+
+		if (this.gen !== 4 && move.selfdestruct === 'always') {
+			this.faint(pokemon, pokemon, move);
+		}
+
+		let damage: number | false | undefined | '' = false;
+		if (move.target === 'all' || move.target === 'foeSide' || move.target === 'allySide' || move.target === 'allyTeam') {
+			damage = this.tryMoveHit(target, pokemon, move);
+			if (damage === this.NOT_FAIL) pokemon.moveThisTurnResult = null;
+			if (damage || damage === 0 || damage === undefined) moveResult = true;
+		} else {
+			if (!targets.length) {
+				this.attrLastMove('[notarget]');
+				this.add(this.gen >= 5 ? '-fail' : '-notarget', pokemon);
+				return false;
+			}
+			if (this.gen === 4 && move.selfdestruct === 'always') {
+				this.faint(pokemon, pokemon, move);
+			}
+			moveResult = this.trySpreadMoveHit(targets, pokemon, move);
+		}
+		if (move.selfBoost && moveResult) this.moveHit(pokemon, pokemon, move, move.selfBoost, false, true);
+		if (!pokemon.hp) {
+			this.faint(pokemon, pokemon, move);
+		}
+
+		if (!moveResult) {
+			this.singleEvent('MoveFail', move, null, target, pokemon, move);
+			return false;
+		}
+
+		if (
+			!move.negateSecondary &&
+			!(move.hasSheerForce && pokemon.hasAbility(['sheerforce', 'aquilasblessing'])) &&
+			!this.getAllActive().some(x => x.hasAbility('skilldrain'))
+		) {
+			const originalHp = pokemon.hp;
+			this.singleEvent('AfterMoveSecondarySelf', move, null, pokemon, target, move);
+			this.runEvent('AfterMoveSecondarySelf', pokemon, target, move);
+			if (pokemon && pokemon !== target && move && move.category !== 'Status') {
+				if (pokemon.hp <= pokemon.maxhp / 2 && originalHp > pokemon.maxhp / 2) {
+					this.runEvent('EmergencyExit', pokemon, pokemon);
 				}
 			}
 		}
```

Comment:
```suggestion
			if (
				pokemon && pokemon !== target &&
				pokemon.hp <= (pokemon.maxhp / 2) && originalHp > (pokemon.maxhp / 2) &&
				move?.category !== 'Status'
			) {
				this.runEvent('EmergencyExit', pokemon, pokemon);
			}
		}
```

Reply:
this is just copypasted from the base scripts.ts, id rather not deal with that